src/inc/engine/render_component.h

```diff
@@ -35,6 +35,12 @@ class RenderComponent
 
   virtual void calculate_destination() = 0;
 
+  template<typename Impl>
+  Impl& as_implementation()
```

Comment:
This is seen as a necessary evil. Different aspects of the game will need to rearrange or manipulate different render components individually. The clients can ask if a `RenderComponent` is of some base `Impl` class. If not, this will throw a `std::bad_cast` and the program will fail.